Startup.cs

```diff
@@ -28,7 +24,7 @@ public void ConfigureServices(IServiceCollection services)
         {
             new Database ().MigrateAsync ().Wait ();
             services.AddScoped<Database, Database> ();
-            services.AddMvc().AddRazorPagesOptions(options =>
+            services.AddRazorPages(options =>

```

Comment:
This also might need `AddController` as well since there are controllers in this project? (Alternatively, `AddMvc()` also still works fine and will add everything)

Reply:
I think I took this from here: https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30?view=aspnetcore-5.0&tabs=visual-studio#mvc-service-registration

> The following example adds support for Razor Pages and minimal controller support.

The controllers do work without an explicit `AddControllers` call, but I'm not opposed to adding it "for safety".

Reply:
Hm, so I just looked it up the differences, and `AddControllers()` does indeed add some features that are not registered otherwise. `AddControllers()` is mostly equivalent to:

```c#
services.AddMvcCore()
    .AddApiExplorer()
    .AddAuthorization()
    .AddCors()
    .AddDataAnnotations()
    .AddFormatterMappings();
```

while `AddRazorPages()` does:

```c#
services.AddMvcCore()
    .AddAuthorization()
    .AddDataAnnotations()
    .AddRazorPages()
    .AddCacheTagHelper();
```

So controllers would be missing the API explorer, CORS and formatter mappings which may or may not be relevant in this project, given that its controllers offers an API. – I think adding `AddControllers()` just to be safe is likely a good call here.